Authentication/SignUp/View/SignupController.swift

```diff
@@ -0,0 +1,230 @@
+            }
+        }
+    }
+    
+    func bindEmailElements() {
+        _viewModel.email <~ signupView.emailTextField.rex_textSignal
+        signupView.emailLabel?.text = _viewModel.emailText
+        signupView.emailTextField.placeholder = _viewModel.emailPlaceholderText
+        _viewModel.emailValidationErrors.signal.observeNext { [unowned self] errors in
+            self._delegate.signupController(self, didFailEmailValidationWithErrors: errors)
+        }
+        if let emailValidationMessageLabel = signupView.emailValidationMessageLabel {
+            emailValidationMessageLabel.rex_text <~ _viewModel.emailValidationErrors.signal.map { $0.first ?? " " }
+        }
+    }
+    
+    func bindPasswordElements() {
+        _viewModel.password <~ signupView.passwordTextField.rex_textSignal
+        signupView.passwordLabel?.text = _viewModel.passwordText
+        signupView.passwordTextField.placeholder = _viewModel.passwordPlaceholderText
+        _viewModel.passwordValidationErrors.signal.observeNext { [unowned self] errors in
+            self._delegate.signupController(self, didFailPasswordValidationWithErrors: errors)
+        }
+        if let passwordValidationMessageLabel = signupView.passwordValidationMessageLabel {
+            passwordValidationMessageLabel.rex_text <~ _viewModel.passwordValidationErrors.signal.map { $0.first ?? " " }
+        }
+        
+        _viewModel.passwordConfirmation <~ signupView.passwordConfirmTextField.rex_textSignal
+        signupView.passwordConfirmLabel?.text = _viewModel.confirmPasswordText
+        signupView.passwordConfirmTextField.placeholder = _viewModel.confirmPasswordPlaceholderText
+        _viewModel.passwordConfirmationValidationErrors.signal.observeNext { [unowned self] errors in
+            self._delegate.signupController(self, didFailPasswordConfirmationValidationWithErrors: errors)
+        }
+        if let passwordConfirmValidationMessageLabel = signupView.passwordConfirmValidationMessageLabel {
+            passwordConfirmValidationMessageLabel.rex_text <~ _viewModel.passwordConfirmationValidationErrors.signal.map { $0.first ?? " " }
+        }
+    }
+    
+    func bindButtons() {
+        signupView.signupButton.setTitle(_viewModel.signupButtonTitle, forState: .Normal)
+        signupView.signupButton.rex_pressed.value = _viewModel.signUpCocoaAction
+        signupView.signupButton.rex_enabled.signal.observeNext { [unowned self] in self.signupView.signupButtonEnabled = $0 }
+        
+        signupView.termsAndServicesLabel?.text = _viewModel.termsAndServicesLabelText
+        signupView.termsAndServicesButton.setTitle(_viewModel.termsAndServicesButtonTitle, forState: .Normal)
+        //signupView.termsAndServicesButton -> Presents the terms and services
+    }
+    
+}
+
+extension SignupController: UITextFieldDelegate {
+    
+    public func textFieldShouldReturn(textField: UITextField) -> Bool {
+        if textField == signupView.usernameTextField {
```

Comment:
I find it cleaner to have an objc associated property `nextTextField`. When we have core, we could add it.
This would end up being
```swift
    textField.nextTextField?. becomeFirstResponder()
```